snuba/datasets/schemas/join.py

```diff
@@ -59,20 +59,21 @@ def get_schemas(self) -> Mapping[str, Schema]:
         raise NotImplementedError
 
 
-@dataclass(frozen=True)
-class TableJoinNode(JoinNode):
+class TableJoinNode(TableSource, JoinNode):
     """
     Represent one qualified data source in the JOIN expression.
     It can be a table or a view.
     """
-    alias: str
-    schema: TableSchema
 
-    def __str__(self) -> str:
-        return f"{self.schema.get_data_source()} {self.alias}"
+    def __init__(self, alias: str, table_name: str, columns: ColumnSet) -> None:
+        super().__init__(table_name, columns)
+        self.__alias = alias
```

Comment:
I don't think it's a good idea for subclasses to change the parameter of their superclass like this.